### DISCUSSION TITLE
AS4RequestHandler: how to change AS4SigningParams?

### DISCUSSION THREAD

**Discussion by sopgreg** [phax/phase4#165] [Q&A]
We discovered an issue where our Receipt messages always used `X509v3` instead of `X509PKIPathv1` in the SecurityTokenReference.

Reason seems to be that in `AS4RequestHandler` there are multiple code paths where the `AS4SigningParams` are created via constructor `new AS4SigningParams` without the ability to modify them externally (by e.g. setting `setUseSingleCertificate(false)`):

Therefore, e.g. (and other places) `AS4RequestHandler#_signResponseIfNeeded` uses these signing params with `setUseSingleCertificate(true)` as default.

![grafik](https://github.com/phax/phase4/assets/33895520/5af23246-1d64-48bb-9244-211e522e5fe3)

Is there something missing to customize the signing params?

**Comment by phax** [comment]
The necessary changes were implemented in #166 and #167 and part of the 2.3.0 release.
That requires you to set the `AS4CryptParams` and `AS4SigningParams` of the `AS4IncomingSecurityConfiguration` object
